docs/src/schema_versioning.md

```diff
@@ -0,0 +1,85 @@
+# Schema versioning
+
+This document describes how schemas produced by fmu-dataio are versioned,
+under what circumstances version numbers change, and provides criteria for the
+kind of version change (patch, minor, major) a schema change causes.
+
+```{note}
+The version of fmu-dataio is decoupled from the versions of schemas it produces.
+This means that a new version of fmu-dataio may cause no schema changes at
+all. Similarly, a new version of fmu-dataio may cause just one schema version
+to change, or even _all_ of them.
+
+There is no direct relationship between an fmu-dataio version and schema
+versions aside from the fact that a schema version may have changed _with_ a
+new fmu-dataio version.
+```
+
+## Schema version changes
+
+Schema versions follow [semantic versioning](http://semver.org/). This gives
+every schema version a specific number in the form `X.Y.Z` where
+
+- `X` is the major version number,
+- `Y` is the minor version number,
+- `Z` is the patch version number (sometimes called the _micro_ version).
+
+Schema version numbers change when a schema is changed. When deciding what
+version a changed schema should become the primary concern should be whether
+or not the change being made is backward compatible. Backwards compatibility
+is broken if metadata generated for and valid against a previous version is
+invalid against the updated version.
```

Comment:
```suggestion
or not the change being made is backward compatible. Backwards compatibility
is broken if some metadata valid with the new schema is not valid against the previous version of the schema.
```
...or some variant of this I think would be preferred definition of "backward compatibility" from webviz point of view (and probably other consumers of data from Sumo and fmu-dataio).